### PR TITLE
add Mock Data

### PR DESCRIPTION
**Pull request by tektechnologies**
adding the weather and location lab from class 5

### CODE REVIEW

**Comment by tektechnologies** [comment]
Correction this is the class 5 demo of the city explorer App, which we are building in class like in 301 of ol'

**Review by selmaT273 — COMMENTED** [review]
There's quite a few places with of wonky indentation/line-spacing/syntax that could use refactoring to make the code more readable. Will continue to look through and add comments though, just sharing my initial observation!

**Comment by tektechnologies** [comment]
right on, I can def go through before merge and clean each file up. 
@selmaT273

**Comment by selmaT273** [comment]
(ope, edited my initial comment)

File: src/views/CityExplorer/CityExplorerMain.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      <h1>Find Your Path</h1>
+      <p>Enter a location in the USA below to learn about the weather, restaurants, movies, and more!</p>
+    </div>
+        <form onSubmit={this.handleLocationSearch}>
+          <label>Find the Fun.</label>
+          <input type="text"  name="search" placeholder="Enter a location." />
+          <button type="sumbit">Find It!</button>
+        </form>
+        {/**add condtional rendering for search */}
+        { this.state.displaySearchResults &&
```
Comment (selmaT273) [inline src/views/CityExplorer/CityExplorerMain.js:46]: I've noticed you used this syntax in several places (across multiple files) and am wondering how this works? I'm not familiar with this syntax.

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: I am guessing you are referring to the {} and the conditional rendering of the component view based on if state is true and the user defined components, restaurant, location, map? The {} allow for js and I think I need them for the comment as well in the render. @selmaT273

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: This we need to make sure that students fully understand, condition ? true : false.

Comment (selmaT273) [reply src/views/CityExplorer/CityExplorerMain.js:46]: Ah, ok. I see that the React docs in that link has the `&&` operator in an example for conditionally including an element/component, but now I am wondering if it is standard practice in React to use this syntax for an if condition, as opposed to using a ternary? 

I agree, the students need to understand how to use ternary syntax (condition ? true : false) as it is specified in the curriculum, but I don't believe the way its written here on lines 43-60 is an example of a ternary as there is no true or false, it only states the condition (missing colon and question mark). I can see the conditional syntax used in these lines as potentially being confusing for newer developers.

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: Makes sense I will use ternary ops, for the rest of the labs. @selmaT273

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: Although hopefully the students are not looking at my code.

Comment (selmaT273) [reply src/views/CityExplorer/CityExplorerMain.js:46]: If you prefer to use an inline if with && instead of ternary, I'm definitely up for chatting more about why and think it could be useful to show students both ways. And yes while I don't necessarily think they'll look at this repo, I was moreso thinking the code you're writing out here is reflective of your preference and/or what you plan to loosely base your demo off of.

Comment (dahlbyk) [reply src/views/CityExplorer/CityExplorerMain.js:46]: To the original question: `condition && <MyComponent />` is a super common React idiom to only render `MyComponent` if `condition` is truthy, i.e. only render search results if there has been a search query. This is exactly how I would have written it.

Reinforcing the ternary conditional (`condition ? ifTrue : ifFalse`) is _also_ important, and the two syntaxes are closely related. The `&&` above is also identical to `condition ? <MyComponent /> : condition`, i.e. the value of the expression is `condition` if `condition` evaluates as falsey (false, null, undefined, 0, ""). The quirk is React chooses not to render some falsey things (`false`, `null`, `""`), renders others (`0`), and complains about `undefined`. The [React Conditional Rendering link](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) includes a section on how you get `<div>0</div>` due to this, when what you probably wanted is not to show a 0.

All of this is stuff we need to understand as instructors, and then choose how deep down the rabbit hole to go in explaining to the class. I've definitely "accidentally" had a 0 show up on the page before with `&&` to clarify that particular quirk, which isn't necessarily obvious.

TL;DR: React devs tend to take advantage of how `||` and `&&` and `? ... :` work in JS, and this seems like a good lecture/demo to reinforce all of that.

> Although hopefully the students are not looking at my code.

They might! Regardless, you should both be practicing how you would "perform", i.e. good variable names, indentation, component decomposition, etc.

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: So the ```<div>0</div>  ``` rendering is the return of the false condition being rendered and react chooses 0 as the value? @dahlbyk

File: src/views/HomePage.js [hunk 2]
```diff
@@ -25,6 +25,17 @@ class HomePage extends React.Component {
                 </Jumbotron>
               </Col>
             </Row>
+            <Card style={{ width: '18rem' }}>
+            <Card.Img variant="top" src="https://picsum.photos/id/281/200/175" />
```
Comment (dahlbyk) [inline src/views/HomePage.js]: This is a good example of indentation that makes it harder to read. `Card.Img`/`Card.Body` belong inside the `Card`, but that's not clear from indentation.

File: src/views/CityExplorer/CityExplorerMain.js (hunk 1 above)
Comment (dahlbyk) [reply src/views/CityExplorer/CityExplorerMain.js:46]: > So the `<div>0</div> ` rendering is the return of the false condition being rendered and react chooses 0 as the value? @dahlbyk

I wouldn't say React "chooses 0 as the value". It's the value because that's how JS works, and then React decides if/how to render the value.

Two steps:
1. JS evaluates the `&&`, which results in the left expression if it's falsey (e.g. 0).
2. React renders the result of the expression in `{...}`, which is `0`. Since 0 is a useful thing to display to the screen, React renders the 0 inside the `<div>`.
    - React _will not_ render `false` or `null` values, which are also falsey. This is really just a quirk of `0` being falsey but still rendered by React.

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: so does the 0 come from the empty object being returned, or from the first half of the condition where the length did not equate to greater than 0? @dahlbyk

Comment (tektechnologies) [reply src/views/CityExplorer/CityExplorerMain.js:46]: or is that just JS and react being react?

**Comment by tektechnologies** [comment]
@selmaT273 I think the indentation is updated, thanks again, let me know if I missed something. thanks.

File: src/views/App.js [hunk 3]
```diff
@@ -21,8 +22,6 @@ class App extends React.Component {
     return(
       <div id="app">
         <Header />
-       
-
                <main>
```
Comment (selmaT273) [inline src/views/App.js]: main is next to header, so they should line up. (same with footer below)

File: src/views/CityExplorer/Weather.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+export class Weather extends Component {
+  
+  render() {
+    console.log('props weather: ', this.props.weather['city_name']);
+    return (
+      <div>
+      <h3>Weather in {this.props.location.search_query}</h3>
+        {this.props.weather &&
+        <div>
```
Comment (selmaT273) [inline src/views/CityExplorer/Weather.js]: this div is next to the h3 and should not be indented

File: src/views/HomePage.js [hunk 5]
```diff
@@ -25,6 +25,17 @@ class HomePage extends React.Component {
                 </Jumbotron>
               </Col>
             </Row>
+            <Card style={{ width: '18rem' }}>
+              <Card.Img variant="top" src="https://picsum.photos/id/281/200/175" />
+                <Card.Body>
+                  <Card.Title>City Explorer</Card.Title>
+                    <Card.Text>
```
Comment (selmaT273) [inline src/views/HomePage.js]: several things in this block that are off on indentation, appearing as if they're nested when they're not

**Review by selmaT273 — APPROVED** [review]
Several issues with indentation/syntax still remaining, left comments pointing a few of them out. Recommend combing through the code base to clean up where things are off before merging.

File: src/views/CityExplorer/Weather.js (hunk 4 above)
Comment (dahlbyk) [reply src/views/CityExplorer/Weather.js]: > this div is next to the h3 and should not be indented

It actually should be indented, because it's inside the `{this.props.weather ...}` expression. Idiomatic React formatting would look like this:

```js
      <div>
        <h3>Weather in {this.props.location.search_query}</h3>
        {this.props.weather &&
          <div>
          </div>
        }
      </div>
```

Thanks for another great example of inconsistent indentation making code harder to read. 😀
